### PR TITLE
update FOX regex to include Fox Sports replays

### PR DESCRIPTION
**Pull request by ischmidt20**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Fox has recently moved sports replays away from fox.com/watch and to foxsports.com/replays. This change updates the extractor to match these new URLs. Nothing in the extractor internals changes.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🦊📺🔧

Improve FOX extractor by expanding URL pattern and supporting more video types. Update `yt_dlp/extractor/fox.py` accordingly.

> _`_VALID_URL` grows_
> _To catch more foxes in watch_
> _Autumn of content_

### Walkthrough
* Update the URL pattern to support more FOX videos ([link](https://github.com/yt-dlp/yt-dlp/pull/7724/files?diff=unified&w=0#diff-cfd20c284ecf185af556c1561729317e968a86ab7af5dda63f53343104955afdL23-R23))

</details>

### CODE REVIEW

File: yt_dlp/extractor/fox.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 
 class FOXIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?fox\.com/watch/(?P<id>[\da-fA-F]+)'
+    _VALID_URL = r'https?://(?:www\.)?fox(?:sports)?\.com/(?:watch|replay)/(?P<id>[\da-fA-F]+)'
```
Comment (bashonly) [inline yt_dlp/extractor/fox.py:23]: could also do it this way, though I don't think it matters
```suggestion
    _VALID_URL = r'https?://(?:www\.)?fox(?:\.com/watch|sports\.com/replay)/(?P<id>[\da-fA-F]+)'
```

Comment (ischmidt20) [reply yt_dlp/extractor/fox.py:23]: I'd rather keep it this way in case a foxsports.com/watch or fox.com/replay appears

**Comment by pukkandan** [comment]
Pls add a test (only_matching is fine)

**Comment by ischmidt20** [comment]
Test added

**Pull request merged**